cmd/bcpd/app/app_test.go

```diff
@@ -680,19 +688,27 @@ func benchmarkSendTx(b *testing.B, nbAccounts, blockSize int) {
 
 	b.ResetTimer()
 
+	// iterate through Txs
+	// start at one to not trigger block creation at the first iteration
+	height := 0
 	for i := 1; i <= b.N; i++ {
 		chres := myApp.CheckTx(txs[i-1])
 		require.Equal(b, uint32(0), chres.Code, chres.Log)
 
-		if i%blockSize == 0 {
-			header := abci.Header{Height: int64(i/blockSize) + 1}
+		// if there is no remainder we have enough txs to create a block
+		// If the number of tx is not a multiple of the block size, we create
+		// a final block at the end with the remaining txs
+		if i%blockSize == 0 || (i == b.N && b.N%blockSize != 0) {
```

Comment:
Good update the to guard:
`i%blockSize == 0 || (i == b.N && b.N%blockSize != 0)`

But you can simplify this:
`i%blockSize == 0 || i == b.N`

If `b.N%blockSize == 0` it will trigger the first point anyway, so no need to exclude it from the second, it will never make it there